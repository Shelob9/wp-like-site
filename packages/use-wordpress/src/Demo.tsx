import React, { useEffect, useState } from 'react';
import useWordPress from './useWordPress';

export const Demo = (props: { page: number }) => {
  const [posts, setPosts] = useState<Array<any>>([]);
  const { wp } = useWordPress('https://ninjaforms.com/wp-json');
  useEffect(() => {
    wp.posts()
      .page(props.page)
      .then((data: Array<any>) => {
        console.log(data);
        setPosts(data);
      })
      .catch((err: Error) => {
        console.log(err);
      });
  }, [setPosts, props.page]);

  return (
    <div>
      <h1>Demo</h1>
      <h2>Posts</h2>
      {posts.length && (
        <ul>
          {posts.map((post: { id: number; title: { rendered: string } }) => (
            <li key={post.id}>{post.title}</li>
          ))}
        </ul>
      )}
    </div>
  );
};
