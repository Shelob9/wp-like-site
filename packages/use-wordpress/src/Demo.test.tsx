import { Demo } from './Demo';
import ReactDOM from 'react-dom';
import React from 'react';
import { render } from '@testing-library/react';

describe('Demo of hook', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Demo page={1} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshots ', () => {
    const { container } = render(<Demo page={1} />);
    expect(container).toMatchSnapshot();
  });
});
