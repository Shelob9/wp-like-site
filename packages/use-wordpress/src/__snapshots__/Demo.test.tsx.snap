// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo of hook Snapshots  1`] = `
<div>
  <div>
    <h1>
      Demo
    </h1>
    <h2>
      Posts
    </h2>
    0
  </div>
</div>
`;
